docs/migration/migrate_to_new_azure_service_broker.md

```diff
@@ -0,0 +1,108 @@
+6641fdfb-abef-4303-ad29-f055c1ca66cc
+```
+
+>>Tip: The CF CLI command `cf service <instance-name> --guid` can check the GUID of a service instance.
+
+For the azure-sqldb instances without the list, the migration script will treat them as existing server case.
+
+### 2. Purge services offering via CF CLI
+
+As above listed services are not able to migrate. We need to purge them via CF CLI. run:
+
+```
+$ cf purge-service-offering azure-mysqldb -f
+$ cf purge-service-offering azure-postgresqldb -f
+$ cf purge-service-offering azure-cosmosdb -f
+$ cf purge-service-offering azure-documentdb -f
+```
+
+>>**WARNING**: `cf purge-service-offering SERVICE` assumes that the service broker responsible for this service offering is no longer available, and all service instances have been deleted. The resources of these instances are still on Azure. You can manually migrate them.
+
+### 3. Install Open Service Broker for Azure
+
+Follow the [doc](https://github.com/Azure/open-service-broker-azure/tree/master/contrib/cf) to install OSBA.
+
+### 4. Migrate service instances records from MASB database(the SQL one) to OSBA database(the redis one).
+
+1. Download migration scripts:
+
+```
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker.sh
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker.js
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker_helper_pc.go
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker_helper_bc.go
+```
+
+2. Get the `manifest.yml`(filled for `cf push`) of both MASB and OSBA.
+
+3. Run:
+
+```
+$ chmod +x migrate_to_new_azure_service_broker.sh
+$ ./migrate_to_new_azure_service_broker.sh <path-to-masb-manifest> <path-to-osba-manifest> (<path-to-new-sql-server-list>)
+```
+
+The broker database migration succeeds if the script ends with no error message.
+
+### 4. Update service broker via CF CLI
+
+Run:
+
+```
+$ cf update-service-broker <service-broker-name> <admin-to-new-broker> <password-to-new-broker> <url-to-new-broker>
+```
+
+The `<service-broker-name>` can be checked by `cf service-brokers` if you forget which name meta-azure-service-broker was registered with.
+
+## Notes (IMPORTANT)
+
+* The bound apps possiblely need some code changes, as the format of credentials delivered by the broker have some changes. Please refer to [open-service-broker-azure docs](https://github.com/Azure/open-service-broker-azure/tree/master/docs/modules) and [meta-azure-service-broker docs](https://github.com/Azure/meta-azure-service-broker/tree/master/docs) to see the differences module-by-module in detail. Note that the credentials just change in new bindings. Credentials in existing bindings won't change.
```

Comment:
Should users `unbind` and `bind` the service instance?

Reply:
If their apps have already done the code change for the new credentials provided by OSBA, they should. Or, they shouldn't. Will add this.

Reply:
Did you add it?

Reply:
Added it now.